### PR TITLE
Add the possibility to listen on a specific host

### LINKED ISSUE #1798: Specifying a bind address does not work
**Describe the bug**
The instructions given for binding to a specific address (--listen \<address\>:\<port\>) do not appear to work. 

From the [Quick Start instructions](https://github.com/novnc/noVNC):
- **"If you don't need to expose the web server to public internet, you can bind to localhost:"**
\
`./utils/novnc_proxy --vnc localhost:5901 --listen localhost:6081`"

It does not bind to that address, and the resulting URL given when it is used this way doesn't make sense.

**Server:**
 - noVNC version: 1.4, and git
 - WebSocket proxy: websockify

### PR DESCRIPTION
**Pull request by afbjorklund**
For instance, for listening only on "localhost"

That is, bind on 127.0.0.1 instead of 0.0.0.0

Closes #1798

### CODE REVIEW

File: docs/novnc_proxy.1 [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
 .SH SYNOPSIS
-.B novnc_proxy [--listen PORT] [--vnc VNC_HOST:PORT] [--cert CERT] [--ssl-only]
+.B novnc_proxy [--listen [HOST:]PORT] [--vnc VNC_HOST:PORT] [--cert CERT] [--ssl-only]
 
 Starts the WebSockets proxy and a mini-webserver and
 provides a cut-and-paste URL to go to.
 
-    --listen PORT         Port for proxy/webserver to listen on
+    --listen [HOST:]PORT  Port for proxy/webserver to listen on
                           Default: 6080
```
Comment (kalvdans) [inline docs/novnc_proxy.1:12]:
```suggestion
                          Default: 6080 on all interfaces
```

File: utils/novnc_proxy [hunk 2]
```diff
@@ -191,9 +200,9 @@ else
     fi
 fi
 
-echo "Starting webserver and WebSockets proxy on port ${PORT}"
-#${HERE}/websockify --web ${WEB} ${CERT:+--cert ${CERT}} ${PORT} ${VNC_DEST} &
-${WEBSOCKIFY} ${SYSLOG_ARG} ${SSLONLY} ${FILEONLY_ARG} --web ${WEB} ${CERT:+--cert ${CERT}} ${KEY:+--key ${KEY}} ${PORT} ${VNC_DEST} ${HEARTBEAT_ARG} ${IDLETIMEOUT_ARG} ${RECORD_ARG} ${TIMEOUT_ARG} ${WEBAUTH_ARG} ${AUTHPLUGIN_ARG} ${AUTHSOURCE_ARG} &
+echo "Starting webserver and WebSockets proxy on host ${HOST} port ${PORT}"
```
Comment (kalvdans) [inline utils/novnc_proxy:203]:
```suggestion
echo "Starting webserver and WebSockets proxy on port $LISTEN"
```

Your suggestion is slightly incorrect. The default setting listens to all interfaces but the message will tell you it listens on a single interface.

File: utils/novnc_proxy [hunk 3]
```diff
@@ -107,8 +109,15 @@ while [ "$*" ]; do
...
+if [ "$LISTEN" != "$PORT" ]; then
+    HOST=${LISTEN%:*}
+    PORT=${LISTEN##*:}
+    # if no host was given, restore hostname
+    [ "$HOST" = "$PORT" ] && HOST=$(hostname)
+fi
+
 # Sanity checks
-if bash -c "exec 7<>/dev/tcp/localhost/${PORT}" &> /dev/null; then
+if bash -c "exec 7<>/dev/tcp/${HOST}/${PORT}" &> /dev/null; then
```
Comment (kalvdans) [inline utils/novnc_proxy:120]: Just disable the sanity check if we give the `--listen` option. Then we can remove the $HOST variable altogether.

**Comment by kalvdans** [comment]
It's more complicated that it needs to be, but it is fine that it just passes the argument unmodified to websockify, and the complications are only for pretty-printing.

LGTM besides my small remarks.

File: utils/novnc_proxy [hunk 4]
```diff
@@ -191,9 +202,9 @@ else
     fi
 fi
 
-echo "Starting webserver and WebSockets proxy on port ${PORT}"
-#${HERE}/websockify --web ${WEB} ${CERT:+--cert ${CERT}} ${PORT} ${VNC_DEST} &
-${WEBSOCKIFY} ${SYSLOG_ARG} ${SSLONLY} ${FILEONLY_ARG} --web ${WEB} ${CERT:+--cert ${CERT}} ${KEY:+--key ${KEY}} ${PORT} ${VNC_DEST} ${HEARTBEAT_ARG} ${IDLETIMEOUT_ARG} ${RECORD_ARG} ${TIMEOUT_ARG} ${WEBAUTH_ARG} ${AUTHPLUGIN_ARG} ${AUTHSOURCE_ARG} &
+echo "Starting webserver and WebSockets proxy on${HOST:+ host ${HOST}} port ${PORT}"
+#${HERE}/websockify --web ${WEB} ${CERT:+--cert ${CERT}} ${LISTEN} ${VNC_DEST} &
```
Comment (kalvdans) [inline utils/novnc_proxy:206]: Remove commented out lines

File: utils/novnc_proxy (hunk 3 above)
Comment (afbjorklund) [reply utils/novnc_proxy]: You will still get an error (exception):
```
OSError: [Errno 98] Address already in use
Failed to start WebSockets proxy
```

**Review by CendioOssman — APPROVED** [review]
Looks good to me. Nice work @afbjorklund!

And thanks for the help with reviewing, @kalvdans.

**Pull request merged**